recipes/icu/all/conanfile.py

```diff
@@ -72,8 +77,8 @@ def layout(self):
         basic_layout(self, src_folder="src")
 
     def package_id(self):
-        if self.options.dat_package_file:
-            dat_package_file_sha256 = sha256sum(str(self.options.dat_package_file))
+        if self.info.options.dat_package_file:
+            dat_package_file_sha256 = sha256sum(str(self.info.options.dat_package_file))
```

Comment:
```suggestion
            dat_package_file_sha256 = _sha256sum(str(self.info.options.dat_package_file))
```

I would suggest adding a check in `validate()` to check that path. So far, in case it's an invalid path, it would fail only in middle of the build() step.